Boyer-Moore/BoyerMoore.swift

```diff
@@ -6,33 +6,35 @@
   http://www.drdobbs.com/database/faster-string-searches/184408171
 */
 extension String {
-    func indexOf(pattern: String) -> String.Index? {
+    func index(of pattern: String, usingHorspoolImprovement: Bool = false) -> Index? {
+        // There are no possible match in an empty string
+        guard !isEmpty else { return nil }
```

Comment:
Same as before, the latter `guard` statement should cover this case just as well.